pkg/identity/types.go

```diff
@@ -13,11 +13,45 @@ const (
 
 // ServiceIdentity is the type used to represent the identity for a service
 // For Kubernetes services this string will be in the format: <ServiceAccount>.<Namespace>.cluster.local
-type ServiceIdentity string
+type ServiceIdentity struct {
+	ServiceAccount string
+	Namespace      string
+	ClusterDomain  string
```

Comment:
In the future, we will need to support non k8s env such as VMs. We most likely will continue to use a ServiceAccount to map non k8s identity types (such as service principals).
```suggestion
	TrustDomain  string
```